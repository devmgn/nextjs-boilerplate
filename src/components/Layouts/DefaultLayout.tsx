import Footer from '../Footer';
import Header from '../Header';

export type DefaultLayoutProps = {
  children: React.ReactNode;
};

const DefaultLayout: React.FC<DefaultLayoutProps> = ({ children }) => {
  return (
    <>
      <Header />
      <main>{children}</main>
      <Footer />
    </>
  );
};

export default DefaultLayout;
