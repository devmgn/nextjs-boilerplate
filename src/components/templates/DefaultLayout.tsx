import Footer from '@/components/organisms/Footer';
import Header from '@/components/organisms/Header';
import DefaultHead from './DefaultHead';

export type DefaultLayoutProps = {
  children: React.ReactNode;
};

const DefaultLayout: React.FC<DefaultLayoutProps> = ({ children }) => {
  return (
    <>
      <DefaultHead />
      <Header />
      <main>{children}</main>
      <Footer />
    </>
  );
};

export default DefaultLayout;
