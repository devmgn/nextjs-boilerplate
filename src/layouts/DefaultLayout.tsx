import Footer from '@/components/organisms/Footer';
import Header from '@/components/organisms/Header';
import DefaultHead from './DefaultHead';

export type DefaultLayoutProps = {
  children: React.ReactNode;
};

const DefaultLayout: React.FC<DefaultLayoutProps> = (props) => {
  return (
    <>
      <DefaultHead />
      <Header />
      <main>{props.children}</main>
      <Footer />
    </>
  );
};

export default DefaultLayout;
