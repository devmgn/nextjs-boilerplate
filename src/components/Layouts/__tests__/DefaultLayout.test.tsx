import { render } from '@testing-library/react';
import DefaultLayout from '../DefaultLayout';

describe('DefaultLayout', () => {
  test('スナップショットテスト', () => {
    const { asFragment } = render(<DefaultLayout>children</DefaultLayout>);
    expect(asFragment()).toMatchSnapshot();
  });
});
