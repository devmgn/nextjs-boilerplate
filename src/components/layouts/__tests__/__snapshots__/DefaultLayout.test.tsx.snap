// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout スナップショットテスト 1`] = `
<DocumentFragment>
  <meta
    charset="utf-8"
  />
  <meta
    content="width=device-width,initial-scale=1.0,minimum-scale=1.0"
    name="viewport"
  />
  <title>
    site name
  </title>
  <meta
    content="default description"
    name="description"
  />
  <link
    href="/images/favicon.ico"
    rel="icon"
  />
  <header>
    header
  </header>
  <main>
    children
  </main>
  <footer>
    footer
  </footer>
</DocumentFragment>
`;
